### PR TITLE
Add simple neural network

### PR DESCRIPTION
**Pull request by CaioCordeiro**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### Tests Results:
```
Trying:
    res = forward_propagation(32, 10000000)
Expecting nothing
ok
Trying:
    res > 31 and res < 33
Expecting:
    True
ok
Trying:
    res = forward_propagation(32, 1000)
Expecting nothing
ok
Trying:
    res > 31 and res < 33
Expecting:
    False
ok
Trying:
    sigmoid_function(3.5)
Expecting:
    0.9706877692486436
ok
Trying:
    sigmoid_function(3.5, True)
Expecting:
    -8.75
ok
1 items had no tests:
    simple_neural_network
2 items passed all tests:
   4 tests in simple_neural_network.forward_propagation
   2 tests in simple_neural_network.sigmoid_function
6 tests in 3 items.
6 passed and 0 failed.
Test passed.
```

### CODE REVIEW

**Comment by CaioCordeiro** [comment]
@CaedenPH Could you check this one? The other one got closed when I was fixing the pre-commit.

**Review by CaedenPH — APPROVED** [review]
Looks good.
As a side note, you could also implement a simple `doctest` check i

**Comment by CaioCordeiro** [comment]
> Looks good. As a side note, you could also implement a simple `doctest` check

Sure, could you show me an example?

**Comment by CaedenPH** [comment]
> > Looks good. As a side note, you could also implement a simple `doctest` check
> 
> Sure, could you show me an example?

Very simple, just ```py
if __name__ == "__main__":
    import doctest

    doctest.testmod()
```
@CaioCordeiro

File: neural_network/simple_neural_network.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+"""
+Forward propagation explanation:
+https://towardsdatascience.com/forward-propagation-in-neural-networks-simplified-math-and-code-version-bbcfef6f9250
+"""
+
+import doctest
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:6]:
```suggestion
```

File: neural_network/simple_neural_network.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        # Error delta
+        layer_1_delta = layer_1_error * sigmoid_function(layer_1, True)
+        # Update weight
+        weight += INITIAL_VALUE * layer_1_delta
+
+    return layer_1 * 100
+
+
+if __name__ == "__main__":
+    print(doctest.testmod())
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:58]:
```suggestion
    import doctest

    doctest.testmod()
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]

File: neural_network/simple_neural_network.py (hunk 1 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py (hunk 2 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        weight += INITIAL_VALUE * layer_1_delta
+
+    return layer_1 * 100
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+    expected = int(input("Expected value: "))
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:60]:
```suggestion
    
    expected = int(input("Expected value: "))
```

**Review by CaedenPH — APPROVED** [review]

**Review by CaedenPH — APPROVED** [review]

**Review by cclauss — APPROVED** [review]

**Pull request merged**